Dockerfile

```diff
@@ -0,0 +1,36 @@
+## Build
+FROM golang:1.13-buster AS build
+
+COPY bigtable /src/bigtable
+COPY storage /src/storage
+
+WORKDIR /src/bigtable/
+RUN go mod download
+RUN go build -o /cbtemulator ./cmd/cbtemulator
+
+WORKDIR /src/storage/
+RUN go mod download
+RUN go build -o /gcsemulator ./cmd/gcsemulator
+
+## Deploy
+FROM gcr.io/distroless/base-debian10 AS cbtemulator
```

Comment:
FWIW, in our other Docker images, we use [`scratch`](https://hub.docker.com/_/scratch) as the base, so that the resulting image is as small as possible. I don't know much about the size of `base-debian10`, but I suspect it's much heavier-weight. And if you don't want/need a shell or other command-line tools in the resulting container, it's perfect.

Reply:
I'll compare using scratch, right now we are around 36MB. the base-debian10 doesnt have shell as well, so they might be similar in resources

Reply:
size reduced to ~17MB, great!

Reply:
BTW, this PR has an example of fixing the directory permissions when using scratch and non-root user in the container:
https://github.com/fullstorydev/hauser/pull/107#pullrequestreview-734434848
But this is for permissions of directories _inside_ the container image.

One key difference between the emulators and hauser is that the emulators almost certainly want to write data to the _host_ file system (so it is preserved across container executions). Since the non-root user in the container will not likely have permissions on the host filesystem, this by default would still likely result in permission errors. So we should add to the documentation that they should use the `--user` argument when starting the container, and run the container as themselves.

Reply:
I fix the problem by creating a folder in the builder image, associating it with the user, and when copying to the final image passing an owner flag, that way the user will have permissions on that folder. It worked in the end!